addons/web/static/tests/views/fields/statusbar_time_field_tests.js

```diff
@@ -0,0 +1,112 @@
+                        {
+                            id: 40,
+                            display_name: "Won",
+                        },
+
+                    ],
+                },
+            },
+        };
+
+        setupViewRegistries();
+    });
+
+    QUnit.module("StatusBarTimeField");
+
+    QUnit.test("statusbar time per stage tests", async function (assert) {
+        await makeView({
+            serverData,
+            type: "form",
+            resModel: "partner",
+            arch: `
+                    <form>
+                        <header>
+                            <field name="json_statustime" invisible="1" />
+                            <field name="stage_id" widget="statusbar_time" />
+                        </header>
+                    </form>`,
+            resId: 1,
+        });
+
+        // 1. check that time button is toggled off and exists
+        let timeButton = target.querySelector("div.o_statusbar_status div div.time-btn-container div.time-btn-hover");
+        assert.notEqual(timeButton, null);
+
+        // 2. toggle on
+        await click(timeButton);
+
+        // 3. check that time is put in the stages
+        let newStage = target.querySelector("div.o_statusbar_status div div.add-arrow button");
+        let newStageFirstSpan = newStage.querySelector("span.me-2");
+        let newStageSecondSpan = newStage.querySelector("span.time-tracking");
+        assert.strictEqual(newStageFirstSpan.innerHTML, "New");
+        assert.strictEqual(newStageSecondSpan.innerHTML, "(1 week)");
+        let qualifiedStage = target.querySelector("div.o_statusbar_status button[data-value='20']");
+        let qualifiedStageFirstSpan = qualifiedStage.querySelector("span.me-2");
+        let qualifiedStageSecondSpan = qualifiedStage.querySelector("span.time-tracking");
+        assert.strictEqual(qualifiedStageFirstSpan.innerHTML, "Qualified");
+        assert.strictEqual(qualifiedStageSecondSpan.innerHTML, "(3 hours)");
+
+        // 4. toggle off
+        timeButton = target.querySelector("div.o_statusbar_status div div.time-btn-container div.time-btn");
+        await click(timeButton);
+
+        // 5. check the time-tracking isn't there anymore
+        newStage = target.querySelector("div.o_statusbar_status div div.add-arrow button");
+        newStageFirstSpan = newStage.querySelector("span.me-2");
+        newStageSecondSpan = newStage.querySelector("span.time-tracking");
+        assert.strictEqual(newStageFirstSpan.innerHTML, "New");
+        assert.strictEqual(newStageSecondSpan, null);
+        qualifiedStage = target.querySelector("div.o_statusbar_status button[data-value='20']");
+        qualifiedStageFirstSpan = qualifiedStage.querySelector("span.me-2");
+        qualifiedStageSecondSpan = qualifiedStage.querySelector("span.time-tracking");
+        assert.strictEqual(qualifiedStageFirstSpan.innerHTML, "Qualified");
+        assert.strictEqual(qualifiedStageSecondSpan, null);
+    });
+})
```

Comment:
you could also check (maybe in another test) that the toggle state is correctly written in/read from localstorage

Reply:
Could you tell me how to access the component state from a test?

Reply:
No need to read the state, you can directly check in the DOM